### PR TITLE
Improve feed versions load

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Companion PR for https://github.com/ibi-group/datatools-server/pull/457.
Partial fix for https://github.com/ibi-group/datatools-server/issues/456.

Basically, the `VersionSelectorDropdown`, `ManagerHeader` and `ReplaceFileFromVersion` components, and related code are set to use the new field `Feed.feedVersionSummaries` which fetches summary data for all feed versions much faster than `Feed.feedVersions`. That field instead is populated on demand as the user browses feed versions.

### CODE REVIEW

**Comment by binh-dam-ibigroup** [comment]
Build failure is going to be resolved by #777.

File: lib/manager/reducers/projects.js [hunk 1]
```diff
@@ -224,7 +224,8 @@ const projects = (state: ProjectsState = defaultState, action: Action): Projects
       }
       return update(state, {
         all: {[projectIndex]: {feedSources: {[sourceIndex]: {$merge: {
-          feedVersions: versions.sort(versionSort)
+          feedVersions: versions.map(() => null), // versions.sort(versionSort), // TODO remove
```
Comment (miles-grant-ibigroup) [inline lib/manager/reducers/projects.js:227]: We can probably remove this now?

**Review by miles-grant-ibigroup — APPROVED** [review]
Works great locally! The extra fetch lag is hardly noticeable. Thanks for putting this together!

File: lib/manager/reducers/projects.js (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/manager/reducers/projects.js]: Yes, thanks, removed in https://github.com/ibi-group/datatools-ui/pull/776/commits/0ce316a2a47261d6690897583c5e05626859e9da.

**Review by philip-cline — APPROVED** [review]
Just one typo I found, otherwise worked well for me!
